homeassistant/components/template/switch.py

```diff
@@ -294,3 +306,88 @@ async def async_turn_off(self, **kwargs: Any) -> None:
         if self._template is None:
             self._state = False
             self.async_write_ha_state()
+
+
+class TriggerSwitchEntity(TriggerEntity, SwitchEntity, RestoreEntity):
+    """Switch entity based on trigger data."""
+
+    domain = SWITCH_DOMAIN
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        coordinator: TriggerUpdateCoordinator,
+        config: ConfigType,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(hass, coordinator, config)
+        name = self._rendered.get(CONF_NAME, DEFAULT_NAME)
+        self._template = config.get(CONF_STATE)
+        if on_action := config.get(CONF_TURN_ON):
+            self.add_script(CONF_TURN_ON, on_action, name, DOMAIN)
+        if off_action := config.get(CONF_TURN_OFF):
+            self.add_script(CONF_TURN_OFF, off_action, name, DOMAIN)
+        self._state: bool | None = None
+        self._attr_assumed_state = self._template is None
+        if not self._attr_assumed_state:
+            self._to_render_simple.append(CONF_STATE)
+            self._parse_result.add(CONF_STATE)
+
+        self._attr_device_info = async_device_info_to_link_from_device_id(
+            hass,
+            config.get(CONF_DEVICE_ID),
+        )
```

Comment:
In a separate PR, we should move this to the base class, it's duplicated in every template entity

Reply:
Sounds good

Reply:
Made a note formyself
https://github.com/home-assistant/core/issues/143663